Integrations/integration-cisco-ise.yml

```diff
@@ -0,0 +1,543 @@
+            }
+        }
+
+    def update_endpoint_by_id(endpoint_id, endpoint_details):
+        """
+        Updates endpoint status
+        """
+        headers = {
+            "Content-Type": "application/vnd.com.cisco.ise.identity.endpoint.1.0+xml; charset=utf-8"
+        }
+        api_endpoint = "/ers/config/endpoint/{}".format(endpoint_id)
+        return http_request('PUT', api_endpoint, {}, endpoint_details, headers)
+
+    def update_endpoint_custom_attribute_command():
+        """
+        corresponds to 'cisco-ise-update-endpoint-custom-attribute' command. Blocks endpoint using predefined custom fields
+        """
+
+        endpoint_id = demisto.args().get('id')
+        endpoint_mac_address = demisto.args().get('macAddress')
+
+        if endpoint_mac_address and not is_mac(endpoint_mac_address):
+            return "Please enter a valid mac address"
+
+        if not endpoint_id and not endpoint_mac_address:
+            return 'Please enter either endpoint id or endpoint mac address'
+
+        if endpoint_mac_address and not endpoint_id:
+            endpoint_id = get_endpoint_id(endpoint_mac_address).get('SearchResult', {}).get('resources', [])[0].get('id', None)
+
+        endpoint_details = get_endpoint_details(endpoint_id)
+
+        if "ERSEndPoint" not in endpoint_details:
+            return 'Failed to get endpoint %s' % endpoint_id
+
+        attribute_names = demisto.args().get('attributeName').split(',')
+        attribute_values = demisto.args().get('attributeValue').split(',')
+
+        attributes_dic = {}
+        for couple in zip(attribute_names, attribute_values):
+            attributes_dic[couple[0]] = couple[1]
+        try:
+            del endpoint_details['ERSEndPoint']['link']
+            if endpoint_details['ERSEndPoint'].get('customAttributes'):
+                endpoint_details['ERSEndPoint']['customAttributes']['customAttributes'] = attributes_dic
+            else:
+                endpoint_details['ERSEndPoint']['customAttributes'] = {'customAttributes': attributes_dic}
+
+            update_result = update_endpoint_by_id(endpoint_id, endpoint_details)
+            if update_result.status_code != 200:
+                return "Update failed for endpoint " + endpoint_id + ". Please check if the custom fields are defined in the system. Got the following response: " + \
+                    json.dumps(json.loads(update_result.text).get('ERSResponse',{}).get('messages',[]))
+
+            update_json = json.loads(update_result.text)
+
+            updated_fields_dict_list = update_json.get('UpdatedFieldsList', {}).get('updatedField',[])
+
+            if len(updated_fields_dict_list) > 0:
+                updated_fields_string = ' the new custom fields are: ' + json.dumps(updated_fields_dict_list[0].get('newValue'))
+            else:
+                updated_fields_string = ", but the fields that you've tried to update already had that specific value"
```

Comment:
Need to improve message that this is a possible case, because it returns it even when the field does not exist